horovod/torch/sync_batch_norm.py

```diff
@@ -0,0 +1,187 @@
+# Based on https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/_functions.py
+# Modifications copyright 2020 Maka Autonomous Robotic Systems
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from horovod.torch.mpi_ops import allgather_async, allreduce_async, Sum, size, synchronize
+
+from distutils.version import LooseVersion
+
+import torch
+from torch.autograd.function import Function
+import torch.nn.functional as F
+from torch.nn.modules.batchnorm import _BatchNorm
+
+
+# Backward compat for old PyTorch
+if not hasattr(torch.jit, 'unused'):
+    torch.jit.unused = lambda x: x
+
+
+class SyncBatchNorm(_BatchNorm):
+    """
+    Applies synchronous version of N-dimensional BatchNorm.  In this version, normalization
+    parameters are synchronized across workers during forward pass.  This is very useful in
+    situations where each GPU can fit a very small number of examples.
+
```

Comment:
Can you include a comment in the docstring about requiring the tensor to be on GPU?

Reply:
Done